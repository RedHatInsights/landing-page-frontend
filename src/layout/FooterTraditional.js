import React from 'react';
import { RhLogo } from '../components';
import { PageSection } from '@patternfly/react-core';
import './FooterTraditional.scss';

const FooterTraditional = () => (
    <PageSection className='ins-c-footer__traditional pf-m-no-fill'>
        <footer role='contentinfo' className='pf-l-flex pf-m-column pf-m-row-on-lg pf-m-align-items-center-on-lg'>
            <a href='https://www.redhat.com' target='_blank' rel='noopener noreferrer'><RhLogo /></a>
            <div className='ins-c-footer__traditional-nav pf-l-flex pf-m-column pf-m-row-on-lg pf-m-flex-1-on-lg pf-m-justify-content-flex-end-on-lg'>
                <p className='copyright pf-m-spacer-xl-on-lg'>Copyright Â© 2019 Red Hat, Inc.</p>
                <nav>
                    <ul className='pf-l-flex pf-m-column pf-m-row-on-md'>
                        <li>
                            <a className='nav-link' href='https://www.redhat.com/en/about/privacy-policy'>
                                Privacy Policy
                            </a>
                        </li>
                        <li>
                            <a className='nav-link' href='https://access.redhat.com/help/terms/'>
                                Terms of Use
                            </a>
                        </li>
                        <li>
                            <a className='nav-link' href='https://www.redhat.com/en/about/all-policies-guidelines'>
                                All Policies and Guidelines
                            </a>
                        </li>
                    </ul>
                </nav>
            </div>
        </footer>
    </PageSection>
);

export default FooterTraditional;
