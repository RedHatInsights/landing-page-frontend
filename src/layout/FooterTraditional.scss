@import '~@red-hat-insights/insights-frontend-components/Utilities/_mixins';
@import '~@patternfly/patternfly/sass-utilities/_all';

.ins-p-browser-support {
    .pf-c-title {
        margin-top: var(--pf-global--spacer--md);
        margin-bottom: var(--pf-global--spacer--xs);
    }
    p { margin: var(--pf-global--spacer--xs) 0; }
    li {
        margin-left: var(--pf-global--spacer--md);
        list-style: disc;
    }
}

.ins-c-footer__traditional {
    padding: var(--pf-global--spacer--2xl);
    background-color: var(--pf-c-page__main-section--BackgroundColor);

    svg { @include rem('width', 150px); }

    .ins-p-footer__browser-support {
        padding: 0;
        &:hover {
            text-decoration: underline;
        }
    }

    .ins-c-footer__traditional-nav {
        .copyright {
            color: var(--pf-global--Color--200);
            @include rem('margin-right', 30px);
        }

        * { font-size: var(--pf-global--FontSize--sm); }

        li + li {
            @media screen and (min-width: $pf-global--breakpoint--lg) {
                &::before {
                    content: '|';
                    color: var(--pf-global--Color--200);
                    margin-right: var(--pf-global--spacer--md);
                }
            }
        }
    }
}
