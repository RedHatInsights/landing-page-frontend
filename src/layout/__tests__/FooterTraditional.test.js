import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FooterTraditional from '../FooterTraditional';

describe('Footer Traditional component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<FooterTraditional />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Link: Red Hat Logo is correct', () => {
    const wrapper = mount(<FooterTraditional />);
    const logo = wrapper.find('.ins-p-footer__logo');
    expect(logo.getDOMNode().getAttribute('target')).toBe('_blank');
    expect(logo.getDOMNode().getAttribute('href')).toBe(
      'https://www.redhat.com'
    );
  });

  it('Click browser support link', () => {
    const wrapper = mount(<FooterTraditional />);
    const browserSupport = wrapper.find('.ins-p-footer__browser-support').at(0);
    browserSupport.simulate('click');
    const browserSupportModal = wrapper
      .find('.ins-p-browser-support .pf-c-button')
      .at(0);
    browserSupportModal.simulate('click');
  });
});
