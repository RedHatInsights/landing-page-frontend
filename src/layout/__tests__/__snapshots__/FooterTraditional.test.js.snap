// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Traditional component should render correctly 1`] = `
<Fragment>
  <PageSection
    className="ins-c-footer__traditional pf-m-no-fill"
  >
    <footer
      className="pf-l-flex pf-m-column pf-m-row-on-lg pf-m-align-items-center-on-lg"
      role="contentinfo"
    >
      <a
        className="ins-p-footer__logo"
        href="https://www.redhat.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <RhLogo />
      </a>
      <div
        className="ins-c-footer__traditional-nav pf-l-flex pf-m-column pf-m-row-on-lg pf-m-flex-1-on-lg pf-m-justify-content-flex-end-on-lg"
      >
        <p
          className="copyright pf-m-spacer-xl-on-lg"
        >
          Copyright © 2019 Red Hat, Inc.
        </p>
        <nav>
          <ul
            className="pf-l-flex pf-m-column pf-m-row-on-md"
          >
            <li>
              <Component
                className="nav-link ins-p-footer__browser-support"
                onClick={[Function]}
                variant="link"
              >
                Browser Support
              </Component>
            </li>
            <li>
              <a
                className="nav-link ins-p-footer__privacy-policy"
                href="https://www.redhat.com/en/about/privacy-policy"
              >
                Privacy Policy
              </a>
            </li>
            <li>
              <a
                className="nav-link ins-p-footer__terms"
                href="https://access.redhat.com/help/terms/"
              >
                Terms of Use
              </a>
            </li>
            <li>
              <a
                className="nav-link ins-p-footer__all-policies"
                href="https://www.redhat.com/en/about/all-policies-guidelines"
              >
                All Policies and Guidelines
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </footer>
  </PageSection>
  <Modal
    actions={Array []}
    appendTo={<body />}
    ariaDescribedById=""
    className="ins-p-browser-support"
    hideTitle={false}
    isFooterLeftAligned={true}
    isLarge={false}
    isOpen={false}
    isSmall={true}
    onClose={[Function]}
    showClose={true}
    title="Browser support"
  >
    <p>
      Red Hat captures and regularly reviews statistical data from our actual web visitors and registered users, rather than generic industry data, to identify the browsers we need to support in alignment with our customers’ needs. Additionally, to safeguard customer data, only browsers which receive security updates from the browser manufacturer are considered for support. We have implemented this policy to ensure that we can provide an excellent experience to a wide user base.
    </p>
    <Title
      headingLevel="h4"
      size="xl"
    >
       Cookies and Javascript 
    </Title>
    <p>
       To successfully interact with our websites and services, your browser must meet the following feature requirements: 
    </p>
    <ul>
      <li>
         The browser must be configured to accept cookies 
      </li>
      <li>
         The browser must be configured to execute JavaScript 
      </li>
    </ul>
    <Title
      headingLevel="h4"
      size="xl"
    >
       Specific browser support 
    </Title>
    <p>
       We validate against and fully support our customers' use of the past two major releases of the following browsers: 
    </p>
    <ul>
      <li>
         Mozilla Firefox 
      </li>
      <li>
         Google Chrome 
      </li>
      <li>
         Apple Safari 
      </li>
      <li>
         Microsoft Edge 
      </li>
    </ul>
  </Modal>
</Fragment>
`;
